### Webtoon 회차별 정보
- 웹툰명 : title
- 회차 : no
- 웹툰 회차의 기본키(ID값)
- 회차별 썸네일링크
- 회차별 평점

In [1]:
import requests
from bs4 import BeautifulSoup
import re

In [2]:
# 함수화
def naver_webtoon(genre):
    
    dic = {}
    dic['unique_id'] = []
    dic['title'] = []
    dic['author'] = []
    dic['total_score'] = []
    dic['main_link'] = []
    dic['image_link'] = []
    dic['update_date'] = []
    
    # first url for title
    # get title with full name
    url = "https://comic.naver.com/webtoon/genre.nhn?view=list&order=ViewCount&genre=" + genre
    response = requests.get(url)
    dom = BeautifulSoup(response.content, "html.parser")
    keywords = dom.select("#content > div.list_area.table_list_area > table > tbody > tr")
    for keyword in keywords:
        dic['title'].append(keyword.select_one("td > a > strong").text)
    
    
    # second url for the others
    url =  "https://comic.naver.com/webtoon/genre.nhn?genre=" + genre
    response = requests.get(url)
    dom = BeautifulSoup(response.content, "html.parser")
    keywords = dom.select(".list_area > .img_list > li")


    # author, total score
    for keyword in keywords:
        dic['author'].append(keyword.select_one("dl > dd > a").text)
        dic['total_score'].append(keyword.select_one("dl > dd > .rating_type > strong").text)
        dic['update_date'].append(keyword.select_one("dl > dd.date2").text)

    # unique id 
    unique_id_list = dom.select("dl > dt > a")
    for i in range(len(unique_id_list)):
        idx = unique_id_list[i].get('href').index("=")
        dic['unique_id'].append(unique_id_list[i].get('href')[idx+1:])

    # main link
    main_link_list = dom.select("dl > dt > a")
    for i in range(len(main_link_list)):
        dic['main_link'].append("https://comic.naver.com" + main_link_list[i].get('href'))

    # image link
    image_link_list = dom.select(".list_area > .img_list > li > .thumb > a > img:nth-of-type(1)")
    for i in range(len(image_link_list)):
        dic['image_link'].append(image_link_list[i].get('src'))
        
    df = pd.DataFrame(dic)
    df[genre] = genre
    
    return df

In [3]:
genres = ["episode","omnibus","story","daily","comic","fantasy","action","drama","pure","sensibility","thrill","historical","sports"]

In [4]:
df = naver_webtoon("episode")

In [5]:
for genre in genres[1:]:
    print(genre)
    sample = naver_webtoon(genre)
    df = pd.merge(df,sample, how='outer')
df.reset_index(inplace=True, drop=True)
df.tail()

omnibus
story
daily
comic
fantasy
action
drama
pure
sensibility
thrill
historical
sports


unique_id      title    author total_score  \
741    703840  한국만화 1990      웹툰작가        9.75   
742    336945    길에서 만나다  쥬드 프라이데이        9.75   
743    515883  사또(Satto)       최윤진        9.80   
744    347685  슈퍼트리오 시즌2       황미나        7.36   
745    716164  로베스의 완전감각       정샛별        9.88   

                                             main_link  \
741  https://comic.naver.com/webtoon/list.nhn?title...   
742  https://comic.naver.com/webtoon/list.nhn?title...   
743  https://comic.naver.com/webtoon/list.nhn?title...   
744  https://comic.naver.com/webtoon/list.nhn?title...   
745  https://comic.naver.com/webtoon/list.nhn?title...   

                                            image_link update_date episode  \
741  https://shared-comic.pstatic.net/thumb/webtoon...  2018.05.17     NaN   
742  https://shared-comic.pstatic.net/thumb/webtoon...  2013.01.17     NaN   
743  https://shared-comic.pstatic.net/thumb/webtoon...  2015.12.29     NaN   
744  https://shared-comic.pstatic.net/thumb/webtoon...  2012.06.04     NaN   
745  https://shared-comic.pstatic.net/thumb/webtoon...  2018.09.04     NaN   

    omnibus  story daily  comic  fantasy  action  drama pure  sensibility  \
741     NaN  story   NaN    NaN      NaN     NaN    NaN  NaN          NaN   
742     NaN  story   NaN    NaN      NaN     NaN  drama  NaN  sensibility   
743     NaN  story   NaN    NaN  fantasy     NaN  drama  NaN          NaN   
744     NaN  story   NaN  comic      NaN     NaN  drama  NaN          NaN   
745     NaN  story   NaN    NaN  fantasy  action    NaN  NaN          NaN   

    thrill historical sports  
741    NaN        NaN    NaN  
742    NaN        NaN    NaN  
743    NaN        NaN    NaN  
744    NaN        NaN    NaN  
745    NaN        NaN    NaN

In [6]:
def make_genre(df):
    genres = ["episode","omnibus","story","daily","comic","fantasy","action","drama","pure","sensibility","thrill","historical","sports"]
    df.fillna("", inplace=True)

    result = []
    for i in range(len(df[genres])):
        ls = list(filter(lambda value: value != "", df[genres].loc[i].values))
        result.append(ls)

    df['genre'] = np.array(["/".join(ls) for ls in result])
    df.drop(columns=genres, axis=1, inplace=True)
    return df

In [7]:
df = make_genre(df)
df.tail()

unique_id      title    author total_score  \
741    703840  한국만화 1990      웹툰작가        9.75   
742    336945    길에서 만나다  쥬드 프라이데이        9.75   
743    515883  사또(Satto)       최윤진        9.80   
744    347685  슈퍼트리오 시즌2       황미나        7.36   
745    716164  로베스의 완전감각       정샛별        9.88   

                                             main_link  \
741  https://comic.naver.com/webtoon/list.nhn?title...   
742  https://comic.naver.com/webtoon/list.nhn?title...   
743  https://comic.naver.com/webtoon/list.nhn?title...   
744  https://comic.naver.com/webtoon/list.nhn?title...   
745  https://comic.naver.com/webtoon/list.nhn?title...   

                                            image_link update_date  \
741  https://shared-comic.pstatic.net/thumb/webtoon...  2018.05.17   
742  https://shared-comic.pstatic.net/thumb/webtoon...  2013.01.17   
743  https://shared-comic.pstatic.net/thumb/webtoon...  2015.12.29   
744  https://shared-comic.pstatic.net/thumb/webtoon...  2012.06.04   
745  https://shared-comic.pstatic.net/thumb/webtoon...  2018.09.04   

                       genre  
741                    story  
742  story/drama/sensibility  
743      story/fantasy/drama  
744        story/comic/drama  
745     story/fantasy/action

아래 코드에서 num range는 파악이 가능하다. 즉, 링크에서 href안에 'no=326' 과 같이 표기가 되어있어 이를 10으로 나누어 page수 파악이 가능하다.

In [121]:
dic2 = {}
dic2['title'] = []
dic2['image_link'] = []
dic2['chapter_subtitle'] = []
dic2['chapeter_score'] = []


count = 1
for tid in titleid:
#     url = "https://comic.naver.com/webtoon/list.nhn?titleId="+ str(tid) +"&page=1"
#     response = requests.get(url)
#     dom = BeautifulSoup(response.content, "html.parser")
#     num_check = dom.select("#content > table.viewList > tr > td:nth-of-type(1) > a > img")
    print("webtoon id:",tid, " count: ", count)
    for num in range(1,201):
        
        url = "https://comic.naver.com/webtoon/list.nhn?titleId="+ str(tid) +"&page=" + str(num)
        response = requests.get(url)
        dom = BeautifulSoup(response.content, "html.parser")
        
        
        # 첫번째 링크가 광고인 경우를 제외시켜줘야 한다.
        links = dom.select("#content > table.viewList > tr > td:nth-of-type(1) > a > img")
        if len(links) > 10:
            links = links[1:]        
        # title
        for i in range(len(links)):
            dic2['title'].append(df[df.unique_id == tid].title.values[0])
            # image link
            dic2['image_link'].append(links[i].get('src'))
            # 회차(episode)
            dic2['chapter_subtitle'].append(links[i].get('title'))
        if len(links) < 10:
            break 
    count += 1

webtoon id: 651673  count:  1
webtoon id: 22052  count:  2
webtoon id: 498158  count:  3
webtoon id: 680911  count:  4
webtoon id: 15441  count:  5
webtoon id: 26316  count:  6
webtoon id: 409630  count:  7
webtoon id: 710747  count:  8
webtoon id: 679519  count:  9
webtoon id: 15938  count:  10
webtoon id: 651663  count:  11
webtoon id: 25517  count:  12
webtoon id: 639604  count:  13
webtoon id: 113121  count:  14
webtoon id: 316914  count:  15
webtoon id: 26101  count:  16
webtoon id: 70046  count:  17
webtoon id: 563782  count:  18
webtoon id: 20853  count:  19
webtoon id: 710751  count:  20
webtoon id: 26144  count:  21
webtoon id: 55169  count:  22
webtoon id: 52946  count:  23
webtoon id: 651664  count:  24
webtoon id: 632342  count:  25
webtoon id: 55143  count:  26
webtoon id: 609477  count:  27
webtoon id: 50422  count:  28
webtoon id: 642598  count:  29
webtoon id: 92106  count:  30
webtoon id: 710754  count:  31
webtoon id: 650304  count:  32
webtoon id: 316912  count:  33


webtoon id: 511447  count:  263
webtoon id: 25695  count:  264
webtoon id: 517773  count:  265
webtoon id: 26460  count:  266
webtoon id: 557676  count:  267
webtoon id: 702422  count:  268
webtoon id: 597481  count:  269
webtoon id: 459546  count:  270
webtoon id: 677737  count:  271
webtoon id: 670149  count:  272
webtoon id: 710761  count:  273
webtoon id: 552960  count:  274
webtoon id: 710760  count:  275
webtoon id: 26108  count:  276
webtoon id: 26107  count:  277
webtoon id: 142911  count:  278
webtoon id: 670150  count:  279
webtoon id: 651617  count:  280
webtoon id: 570506  count:  281
webtoon id: 25050  count:  282
webtoon id: 713975  count:  283
webtoon id: 698918  count:  284
webtoon id: 655748  count:  285
webtoon id: 699658  count:  286
webtoon id: 478262  count:  287
webtoon id: 337964  count:  288
webtoon id: 258205  count:  289
webtoon id: 714839  count:  290
webtoon id: 537990  count:  291
webtoon id: 551651  count:  292
webtoon id: 511446  count:  293
webtoon id: 6

webtoon id: 300957  count:  521
webtoon id: 58293  count:  522
webtoon id: 632705  count:  523
webtoon id: 452117  count:  524
webtoon id: 639080  count:  525
webtoon id: 656641  count:  526
webtoon id: 679568  count:  527
webtoon id: 629056  count:  528
webtoon id: 710752  count:  529
webtoon id: 579352  count:  530
webtoon id: 602930  count:  531
webtoon id: 697681  count:  532
webtoon id: 316913  count:  533
webtoon id: 703853  count:  534
webtoon id: 58219  count:  535
webtoon id: 462900  count:  536
webtoon id: 622646  count:  537
webtoon id: 710763  count:  538
webtoon id: 449854  count:  539
webtoon id: 604146  count:  540
webtoon id: 613933  count:  541
webtoon id: 703849  count:  542
webtoon id: 665170  count:  543
webtoon id: 710741  count:  544
webtoon id: 546623  count:  545
webtoon id: 703634  count:  546
webtoon id: 657948  count:  547
webtoon id: 703631  count:  548
webtoon id: 644112  count:  549
webtoon id: 709731  count:  550
webtoon id: 459545  count:  551
webtoon id

In [122]:
for k in dic2.keys():
    print(len(dic2[k]))

187691
187691
187691


In [123]:
df_episode = pd.DataFrame(dic2)
df_episode.tail()

title                                         image_link  \
187686  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
187687  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
187688  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
187689  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
187690  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   

                      episode  
187686     Part 9. 의천검 탈취작전 1  
187687  Part 8. 쌍감자 체포 프로젝트 2  
187688  Part 8. 쌍감자 체포 프로젝트 1  
187689      Part 7. 일천불의 화가 2  
187690      Part 7. 일천불의 화가 1

In [189]:
for k in dic2.keys():
    print(len(dic2[k]))

187691
187691
187691
250460


In [214]:
dic2['episode_count']

['4',
 '3',
 '2',
 '1',
 '10',
 '9',
 '8',
 '7',
 '6',
 '5',
 '4',
 '3',
 '2',
 '1',
 '10',
 '9',
 '8',
 '7',
 '6',
 '5',
 '4',
 '3',
 '2',
 '1',
 '10',
 '9',
 '8',
 '7',
 '6',
 '5',
 '4',
 '3',
 '2',
 '1',
 '10',
 '9',
 '8',
 '7',
 '6',
 '5',
 '4',
 '3',
 '2',
 '1',
 '10',
 '9',
 '8',
 '7',
 '6',
 '5',
 '4',
 '3',
 '2',
 '1',
 '10',
 '9',
 '8',
 '7',
 '6',
 '5',
 '4',
 '3',
 '2',
 '1',
 '16',
 '15',
 '14',
 '13',
 '12',
 '11',
 '10',
 '9',
 '8',
 '7',
 '',
 '6',
 '5',
 '4',
 '3',
 '2',
 '1',
 '12',
 '11',
 '10',
 '9',
 '8',
 '7',
 '6',
 '5',
 '4',
 '3',
 '',
 '2',
 '1',
 '',
 '6',
 '5',
 '4',
 '3',
 '2']

In [251]:
# 중복행 발견
df_copy = df_episode.drop_duplicates() # Dataframe의 중복행 제거
df_copy['image_link'].value_counts()

https://shared-comic.pstatic.net/thumb/webtoon/25455/477/thumbnail_202x120_d18127cf-afce-4f01-9705-6c8393227b8a.jpg     1
https://shared-comic.pstatic.net/thumb/webtoon/440447/15/inst_thumbnail_20120518174006.jpg                              1
https://shared-comic.pstatic.net/thumb/webtoon/22027/19/inst_thumbnail_22027_19.jpg                                     1
https://shared-comic.pstatic.net/thumb/webtoon/55169/46/inst_thumbnail_20090220180708.jpg                               1
https://shared-comic.pstatic.net/thumb/webtoon/602910/63/inst_thumbnail_20150213181318.jpg                              1
https://shared-comic.pstatic.net/thumb/webtoon/316908/7/inst_thumbnail_20110506115337.jpg                               1
https://shared-comic.pstatic.net/thumb/webtoon/642598/229/thumbnail_202x120_e9bf359f-bfe7-4aa9-b1ca-c24d2c6c14e1.jpg    1
https://shared-comic.pstatic.net/thumb/webtoon/597478/165/thumbnail_202x120_307bae39-3a86-4ed2-9c48-e9102f13bc76.jpg    1
https://shared-comic.pst

In [255]:
df_copy['image_link'].value_counts().max()

1

In [257]:
df_copy.shape

(54701, 3)

In [275]:
# episode count 추출
sample = df_copy['image_link'].values[16]
sample

'https://shared-comic.pstatic.net/thumb/webtoon/651673/320/thumbnail_202x120_169987fd-b926-431b-a26f-d9b7819824f0.jpg'

In [276]:
print(sample.split('/'))
sample.split('/')[5], sample.split('/')[6]

['https:', '', 'shared-comic.pstatic.net', 'thumb', 'webtoon', '651673', '320', 'thumbnail_202x120_169987fd-b926-431b-a26f-d9b7819824f0.jpg']


('651673', '320')

In [277]:
df_copy['titleid'] = df_copy['image_link'].map(lambda x : x.split('/')[5] if len(x.split('/')) > 7 else 0)
df_copy['episode_cnt'] = df_copy['image_link'].map(lambda x : x.split('/')[6] if len(x.split('/')) > 7 else 0)
df_copy.tail()

C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.
C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


title                                         image_link  \
187686  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
187687  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
187688  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
187689  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
187690  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   

                      episode titleid episode_cnt  
187686     Part 9. 의천검 탈취작전 1  347685           6  
187687  Part 8. 쌍감자 체포 프로젝트 2  347685           5  
187688  Part 8. 쌍감자 체포 프로젝트 1  347685           4  
187689      Part 7. 일천불의 화가 2  347685           3  
187690      Part 7. 일천불의 화가 1  347685           2

In [283]:
idx = df_copy[df_copy['episode_cnt'] == 0].index
df_copy.drop(index=idx.values, inplace=True)
df_copy.loc[idx,:]

C:\ProgramData\Anaconda3\lib\site-packages\pandas\core\frame.py:3697: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  errors=errors)


Empty DataFrame
Columns: [title, image_link, episode, titleid, episode_cnt]
Index: []

In [285]:
df_copy.tail()

title                                         image_link  \
187686  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
187687  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
187688  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
187689  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
187690  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   

                      episode titleid episode_cnt  
187686     Part 9. 의천검 탈취작전 1  347685           6  
187687  Part 8. 쌍감자 체포 프로젝트 2  347685           5  
187688  Part 8. 쌍감자 체포 프로젝트 1  347685           4  
187689      Part 7. 일천불의 화가 2  347685           3  
187690      Part 7. 일천불의 화가 1  347685           2

In [286]:
df_copy.to_csv("data_Juno/episode_per_webtoon_naver.csv", index=False)

### 코드가 뒤죽박죽이어서 output이 이상하지만, 정리된 코딩 순서는 위와 같고,최종 row의 갯수는 54,701개이다.(187,686개는 중복제거 전 row의 갯수)

---

### 회차별 평점 가져오기

In [1]:
import requests
import json
from bs4 import BeautifulSoup
import time
from IPython.display import clear_output # clear_output() 으로 아웃풋 제거 가능

In [2]:
def make_url(titleid, no, page):
    objectid = str(titleid) + "_" + str(no)
    return "https://apis.naver.com/commentBox/cbox/web_naver_list_jsonp.json?ticket=comic&templateId=webtoon&pool=cbox3&lang=ko&country=KR&objectId=" + str(objectid) + "&categoryId=&pageSize=100&page=" + str(page) + "&sort=NEW"

In [3]:
df = pd.read_csv('data_Juno/episode_per_webtoon_naver.csv')
df.tail()

title                                         image_link  \
54691  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
54692  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
54693  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
54694  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
54695  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   

                     episode  titleid  episode_cnt  
54691     Part 9. 의천검 탈취작전 1   347685            6  
54692  Part 8. 쌍감자 체포 프로젝트 2   347685            5  
54693  Part 8. 쌍감자 체포 프로젝트 1   347685            4  
54694      Part 7. 일천불의 화가 2   347685            3  
54695      Part 7. 일천불의 화가 1   347685            2

In [4]:
df[['titleid','episode_cnt']].values

array([[651673,    336],
       [651673,    335],
       [651673,    334],
       ...,
       [347685,      4],
       [347685,      3],
       [347685,      2]], dtype=int64)

In [5]:
def cal_page(titleid,no,page=1):
    """
    댓글 페이지 수를 계산하기 위해, 
    전체 댓글 수를 계산하는 함수 입니다.
    """
    url = make_url(titleid, no, page)
    header = {
    'Host' : "apis.naver.com",
    'Referer' : "https://comic.naver.com/comment/comment.nhn?titleId=" + str(titleid) + "&no=" + str(no),
    'Content-Type' : "application/javascript"
    }
    response = requests.get(url, headers=header)
#     dom = BeautifulSoup(response.content, 'html.parser')
    
    one = response.text.find('(') + 1
#     two = dom.text.find(');')
    content = json.loads(response.text[one:-2])
    
    comments = content['result']['count']['comment']
    return int(comments)

In [6]:
error = []
ls_all = df[['titleid','episode_cnt']].values
len(ls_all)

54696

In [11]:
total_comments = []
for i in range(len(ls_all)):
    ls = ls_all[i]
    clear_output()
    print('now working :',i, 'nd // titleid : ',ls[0], "no : ", ls[1])
    comments = cal_page(ls[0], ls[1])
    total_comments.append(comments)
df['total_comments'] = total_comments
df.tail()

now working : 54695 nd // titleid :  347685 no :  2


title                                         image_link  \
54691  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
54692  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
54693  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
54694  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
54695  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   

                     episode  titleid  episode_cnt  total_comments  
54691     Part 9. 의천검 탈취작전 1   347685            6             117  
54692  Part 8. 쌍감자 체포 프로젝트 2   347685            5             159  
54693  Part 8. 쌍감자 체포 프로젝트 1   347685            4             161  
54694      Part 7. 일천불의 화가 2   347685            3             162  
54695      Part 7. 일천불의 화가 1   347685            2             775

In [12]:
df.to_csv("data_Juno/episode_per_webtoon_naver.csv", index=False)

### 회차별 정보에 빠진 '까뱅' 웹툰 회차별 정보 가져오기

In [21]:
webtoon = pd.read_csv("data_Juno/webtoon_naver_info_by_genres.csv")
webtoon.tail()

unique_id      title   author  total_score  \
740     622643    진눈깨비 소년  쥬드프라이데이         9.97   
741     703840  한국만화 1990     웹툰작가         9.75   
742     336945    길에서 만나다  쥬드프라이데이         9.75   
743     515883  사또(Satto)      최윤진         9.80   
744     347685  슈퍼트리오 시즌2      황미나         7.36   

                                             main_link  \
740  https://comic.naver.com/webtoon/list.nhn?title...   
741  https://comic.naver.com/webtoon/list.nhn?title...   
742  https://comic.naver.com/webtoon/list.nhn?title...   
743  https://comic.naver.com/webtoon/list.nhn?title...   
744  https://comic.naver.com/webtoon/list.nhn?title...   

                                            image_link update_date  \
740  https://shared-comic.pstatic.net/thumb/webtoon...  2018.08.30   
741  https://shared-comic.pstatic.net/thumb/webtoon...  2018.05.17   
742  https://shared-comic.pstatic.net/thumb/webtoon...  2013.01.17   
743  https://shared-comic.pstatic.net/thumb/webtoon...  2015.12.29   
744  https://shared-comic.pstatic.net/thumb/webtoon...  2012.06.04   

                       genre  
740   story/pure/sensibility  
741                    story  
742  story/drama/sensibility  
743      story/fantasy/drama  
744        story/comic/drama

In [61]:
list(chapter.titleid.unique())[0], len(list(chapter.titleid.unique()))

(15395, 744)

In [63]:
check = list(webtoon.unique_id.unique())

In [64]:
for titleid in chapter.titleid.unique():
    check.pop(check.index(titleid))
check

[22073]

In [65]:
df[df.titleid == 22073]

Empty DataFrame
Columns: [title, image_link, episode, titleid, episode_cnt, total_comments]
Index: []

In [67]:
webtoon[webtoon.title == '까뱅']

unique_id title  author  total_score  \
45      22073    까뱅  X-TEAM         8.79   

                                            main_link  \
45  https://comic.naver.com/webtoon/list.nhn?title...   

                                           image_link update_date  \
45  https://shared-comic.pstatic.net/thumb/webtoon...  2007.02.23   

            genre  
45  episode/comic

'까뱅' 이라는 웹툰 정보는 있으나, 회차별 정보는 가져오지 못함.

In [95]:
kabang = {}
kabang['title'] = []
kabang['image_link'] = []
kabang['episode'] = []
# '까뱅' 웹툰 회차별 정보 추가(concat)
for num in range(1,5):
    url = "https://comic.naver.com/webtoon/list.nhn?titleId=22073&page="+str(num)
    response = requests.get(url)
    dom = BeautifulSoup(response.content, 'lxml')
    keywords = dom.select("#content > table > tr")
    for keyword in keywords:
        image_link = keyword.select("td:nth-of-type(1) > a > img")
        kabang['title'].append('까뱅')
        kabang['image_link'].append(image_link[0].get('src'))
        kabang['episode'].append(image_link[0].get('title'))

In [97]:
for i in kabang.keys():
    print(i, len(kabang[i]))

title 32
image_link 32
episode 32


In [99]:
df_kabang = pd.DataFrame(kabang)
df_kabang.tail()

title                                         image_link  \
27    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
28    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
29    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
30    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
31    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   

                  episode  
27     GGAVANG 5화 <침투(2)>  
28     GGAVANG 4화 <침투(1)>  
29        GGAVANG 3화 <화산>  
30  GGAVANG 2화 <레이저 빔(2)>  
31  GGAVANG 1화 <레이저 빔(1)>

In [100]:
df_kabang['titleid'] = df_kabang['image_link'].map(lambda x : x.split('/')[5] if len(x.split('/')) > 7 else 0)
df_kabang['episode_cnt'] = df_kabang['image_link'].map(lambda x : x.split('/')[6] if len(x.split('/')) > 7 else 0)
df_kabang.tail()

title                                         image_link  \
27    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
28    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
29    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
30    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
31    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   

                  episode titleid episode_cnt  
27     GGAVANG 5화 <침투(2)>   22073           5  
28     GGAVANG 4화 <침투(1)>   22073           4  
29        GGAVANG 3화 <화산>   22073           3  
30  GGAVANG 2화 <레이저 빔(2)>   22073           2  
31  GGAVANG 1화 <레이저 빔(1)>   22073           1

In [102]:
ls_all = df_kabang[['titleid','episode_cnt']].values
len(ls_all)

32

In [103]:
# 각 회차별 댓글 수 column 추가
total_comments = []
for i in range(len(ls_all)):
    ls = ls_all[i]
    clear_output()
    print('now working :',i, 'nd // titleid : ',ls[0], "no : ", ls[1])
    comments = cal_page(ls[0], ls[1])
    total_comments.append(comments)
df_kabang['total_comments'] = total_comments
df_kabang.tail()

now working : 31 nd // titleid :  22073 no :  1


title                                         image_link  \
27    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
28    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
29    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
30    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
31    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   

                  episode titleid episode_cnt  total_comments  
27     GGAVANG 5화 <침투(2)>   22073           5             419  
28     GGAVANG 4화 <침투(1)>   22073           4             384  
29        GGAVANG 3화 <화산>   22073           3             507  
30  GGAVANG 2화 <레이저 빔(2)>   22073           2             577  
31  GGAVANG 1화 <레이저 빔(1)>   22073           1             945

In [104]:
df.tail()

title                                         image_link  \
54691  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
54692  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
54693  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
54694  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   
54695  슈퍼트리오 시즌2  https://shared-comic.pstatic.net/thumb/webtoon...   

                     episode  titleid  episode_cnt  total_comments  
54691     Part 9. 의천검 탈취작전 1   347685            6             117  
54692  Part 8. 쌍감자 체포 프로젝트 2   347685            5             159  
54693  Part 8. 쌍감자 체포 프로젝트 1   347685            4             161  
54694      Part 7. 일천불의 화가 2   347685            3             162  
54695      Part 7. 일천불의 화가 1   347685            2             775

In [109]:
df_episode = pd.concat([df,df_kabang], axis=0).reset_index(drop=True)
df_episode.tail()

title                                         image_link  \
54723    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54724    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54725    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54726    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54727    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   

                     episode titleid episode_cnt  total_comments  
54723     GGAVANG 5화 <침투(2)>   22073           5             419  
54724     GGAVANG 4화 <침투(1)>   22073           4             384  
54725        GGAVANG 3화 <화산>   22073           3             507  
54726  GGAVANG 2화 <레이저 빔(2)>   22073           2             577  
54727  GGAVANG 1화 <레이저 빔(1)>   22073           1             945

In [110]:
df_episode.to_csv("data_Juno/episode_per_webtoon_naver.csv", index=False)

In [111]:
import getpass
from sqlalchemy import *
import sqlalchemy
import pickle
from sqlalchemy.orm import sessionmaker
from sqlalchemy.ext.declarative import declarative_base
import MySQLdb
from sqlalchemy import create_engine

In [112]:
myid = getpass.getpass("input DB id : ")
mypw = getpass.getpass("input DB pw : ")
SQLALCHEMY_DATABASE_URI = 'mysql://'+myid+':'+mypw+'@rds-juno.civpw0n2c5wo.ap-northeast-2.rds.amazonaws.com/naver_webtoon?charset=utf8mb4'

input DB id : ········
input DB pw : ········


In [113]:
engine = sqlalchemy.create_engine(SQLALCHEMY_DATABASE_URI,encoding='utf-8')

In [114]:
df_episode.to_sql("webtoon_chapters", engine, if_exists='append')

### 회차별 평점 가져오기
- 회차별 총 회차수 / 10 (+1) 로 페이지 수 check

업데이트로 인해 회차 페이지로 들어가서 정보를 가져온다.

In [117]:
df_episode.tail()

title                                         image_link  \
54723    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54724    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54725    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54726    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54727    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   

                     episode titleid episode_cnt  total_comments  
54723     GGAVANG 5화 <침투(2)>   22073           5             419  
54724     GGAVANG 4화 <침투(1)>   22073           4             384  
54725        GGAVANG 3화 <화산>   22073           3             507  
54726  GGAVANG 2화 <레이저 빔(2)>   22073           2             577  
54727  GGAVANG 1화 <레이저 빔(1)>   22073           1             945

In [236]:
ls_all = df_episode[['titleid','episode_cnt']].values
episode_score = []

In [213]:
myid = getpass.getpass("input your id : ")
mypw = getpass.getpass("input your password : ")

input your id : ········
input your password : ········


In [214]:
LOGIN_DATA = {
    'id': myid,
    'pw': mypw, 
}

In [237]:
count = 1
for i, ls in enumerate(ls_all):
    print("now : ", i," titleid : ",ls[0], " episode: ",ls[1])
    if count % 10 ==0:
        clear_output()
    url = "https://comic.naver.com/webtoon/detail.nhn?titleId="+ str(ls[0]) +"&no=" + str(ls[1])
    try:
        response = requests.get(url)
        dom = BeautifulSoup(response.content, 'html.parser')
        score = dom.select_one("#topPointTotalNumber > strong").text
        episode_score.append(score)
    except:
        headers = {
            'cookie':'NNB=B5R2KGOJCPQVU; npic=z0dr3si6GPv0JLs+lOfKo07CMYF0ITC7YCWqlq0KbB4CdszDx1cgXjdJ9NYthOrdCA==; ASID=dc4cf77d0000016301ea74e50000005f; nx_ssl=2; visibleRemocon=false; page_uid=I1cd5BjnXqywEwxYmw14xw==; increase="22073_22_webtoon,22073_28_webtoon,22073_32_webtoon,716163_2_webtoon,22045_561_webtoon,682265_1_webtoon,"; BMR=s=1536499697860&r=https%3A%2F%2Fnstore.naver.com%2Fcomic%2Fdetail.nhn%3FproductNo%3D2370080&r2=https%3A%2F%2Fcomic.naver.com%2Fsearch.nhn%3Fkeyword%3D3%25EC%259D%25B8%25EC%25B9%25AD; JSESSIONID=FC9972F70F9E523AD3E25D52632152E2; nid_inf=73677094; NID_AUT=P2ddnK4yh1GH4GMQAamj0LT+ATAw3b+Y2LXYcjfbYeTI+H3h6GPlxsmdZkMay8cm; NID_JKL=E9miq8hRz31eUFb+JbK7c3FRFe7fSSizQNR10s4eoyI=; NID_SES=AAABsrBdxbiIY+qErd/RmZZ1r4MUTe1ba1gmVadEycG1NcZk2ku5bi7xJIoYpNm3YTJfveB08YOT50F3mVSR/S/XZMRAV36mMRZ052rm1qxGC8BDTD7sTSfqHC50rpH8KqHai3+Y6Voju7SK3CrMvtVkfjE64g20aq1xDyFDmcD3zr3+lkwK326fOBP3scINuhaCmY1GK6DNP5+5pmwEyCUBZ3IEKOsLg89KZMVvmFuacokPz8ettE0XbWS1UNQ03zcsBeylvVaKfbnBrQfgNmJj/aVA3naqTN88n2DtacSxwXO9AWmnInJWiLa9Uk6GYN3m4d5YP+cNGCKeSNFC0bxOSbhXDLncQqbHo13UyHOiGM7YFj7a/330BGHrlSp70wdtg8ixw5fhhDI5FAMvxsM4Ii5Ry0rcnEbk4NcIJ2LCK/ubzZ89ID9OTg7vSOk9aI6Gu46nOUPnwQ8W6S8+II6iOWba4rVrS4ALNAfDvjEgUALDuP8qJwsqDnKmmQ3Le45Hm0feya3oFAVoIh6LI5EvbUlDS8X4kx18Qj8i6cyfP/lafo6LRJ4QwMGySaOnBuapgM1CXyEkyKejZDbRDgxLXCs='
        }
        response = requests.get(url, headers=headers)
        dom = BeautifulSoup(response.content, 'html.parser')
        score = dom.select_one("#topPointTotalNumber > strong").text
#         print(score)
        episode_score.append(score)
    count += 1
# print(len(episode_score))

now :  54720  titleid :  22073  episode:  8
now :  54721  titleid :  22073  episode:  7
now :  54722  titleid :  22073  episode:  6
now :  54723  titleid :  22073  episode:  5
now :  54724  titleid :  22073  episode:  4
now :  54725  titleid :  22073  episode:  3
now :  54726  titleid :  22073  episode:  2
now :  54727  titleid :  22073  episode:  1


In [238]:
len(episode_score)

54728

In [239]:
df_episode['episode_score'] = episode_score
df_episode.tail()

title                                         image_link  \
54723    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54724    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54725    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54726    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54727    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   

                     episode titleid episode_cnt  total_comments episode_score  
54723     GGAVANG 5화 <침투(2)>   22073           5             419          8.93  
54724     GGAVANG 4화 <침투(1)>   22073           4             384          9.00  
54725        GGAVANG 3화 <화산>   22073           3             507          9.10  
54726  GGAVANG 2화 <레이저 빔(2)>   22073           2             577          8.91  
54727  GGAVANG 1화 <레이저 빔(1)>   22073           1             945          8.91

In [240]:
df_episode.to_csv("data_Juno/episode_per_webtoon_naver.csv", index=False)

### 웹툰 좋아요 수 가져오기

In [241]:
webtoon.tail()

unique_id      title   author  total_score  \
740     622643    진눈깨비 소년  쥬드프라이데이         9.97   
741     703840  한국만화 1990     웹툰작가         9.75   
742     336945    길에서 만나다  쥬드프라이데이         9.75   
743     515883  사또(Satto)      최윤진         9.80   
744     347685  슈퍼트리오 시즌2      황미나         7.36   

                                             main_link  \
740  https://comic.naver.com/webtoon/list.nhn?title...   
741  https://comic.naver.com/webtoon/list.nhn?title...   
742  https://comic.naver.com/webtoon/list.nhn?title...   
743  https://comic.naver.com/webtoon/list.nhn?title...   
744  https://comic.naver.com/webtoon/list.nhn?title...   

                                            image_link update_date  \
740  https://shared-comic.pstatic.net/thumb/webtoon...  2018.08.30   
741  https://shared-comic.pstatic.net/thumb/webtoon...  2018.05.17   
742  https://shared-comic.pstatic.net/thumb/webtoon...  2013.01.17   
743  https://shared-comic.pstatic.net/thumb/webtoon...  2015.12.29   
744  https://shared-comic.pstatic.net/thumb/webtoon...  2012.06.04   

                       genre  
740   story/pure/sensibility  
741                    story  
742  story/drama/sensibility  
743      story/fantasy/drama  
744        story/comic/drama

In [243]:
titleids = webtoon.unique_id.values

In [289]:
like = []
for i, titleid in enumerate(titleids):
    print("now: ", i, " titleid : ",titleid)
    if i % 10 == 0:
        clear_output()
    url = "https://comic.like.naver.com/likeIt/likeItContent.jsonp?_callback=window.__jindo2_callback.$615&serviceId=COMIC&displayId=COMIC&contentsId="+ str(titleid) +"&lang=ko&viewType=like"
    response = requests.get(url)
    one = response.text.find('(') + 1
    content = json.loads(response.text[one:-2])
    like_cnt = content['result']['likeItContent']['likeItCount']
    like.append(like_cnt)

now:  741  titleid :  703840
now:  742  titleid :  336945
now:  743  titleid :  515883
now:  744  titleid :  347685


In [290]:
len(like)

745

In [292]:
webtoon['like_count'] = like
webtoon.tail()

unique_id      title   author  total_score  \
740     622643    진눈깨비 소년  쥬드프라이데이         9.97   
741     703840  한국만화 1990     웹툰작가         9.75   
742     336945    길에서 만나다  쥬드프라이데이         9.75   
743     515883  사또(Satto)      최윤진         9.80   
744     347685  슈퍼트리오 시즌2      황미나         7.36   

                                             main_link  \
740  https://comic.naver.com/webtoon/list.nhn?title...   
741  https://comic.naver.com/webtoon/list.nhn?title...   
742  https://comic.naver.com/webtoon/list.nhn?title...   
743  https://comic.naver.com/webtoon/list.nhn?title...   
744  https://comic.naver.com/webtoon/list.nhn?title...   

                                            image_link update_date  \
740  https://shared-comic.pstatic.net/thumb/webtoon...  2018.08.30   
741  https://shared-comic.pstatic.net/thumb/webtoon...  2018.05.17   
742  https://shared-comic.pstatic.net/thumb/webtoon...  2013.01.17   
743  https://shared-comic.pstatic.net/thumb/webtoon...  2015.12.29   
744  https://shared-comic.pstatic.net/thumb/webtoon...  2012.06.04   

                       genre  like_count  
740   story/pure/sensibility       76459  
741                    story       13787  
742  story/drama/sensibility        3564  
743      story/fantasy/drama       21366  
744        story/comic/drama          29

In [296]:
webtoon.to_csv("data_Juno/webtoon_naver_info_by_genres.csv", index=False)

---

### 회차별로 추가할 정보(column)
- 회차별 좋아요
- 회차별 업데이트 시간

In [1]:
import requests, json
from bs4 import BeautifulSoup
from IPython.display import clear_output
import threading
import time

In [2]:
df_episode = pd.read_csv("data_Juno/episode_per_webtoon_naver.csv")
df_episode.tail()

title                                         image_link  \
54723    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54724    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54725    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54726    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54727    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   

                     episode  titleid  episode_cnt  total_comments  \
54723     GGAVANG 5화 <침투(2)>    22073            5             419   
54724     GGAVANG 4화 <침투(1)>    22073            4             384   
54725        GGAVANG 3화 <화산>    22073            3             507   
54726  GGAVANG 2화 <레이저 빔(2)>    22073            2             577   
54727  GGAVANG 1화 <레이저 빔(1)>    22073            1             945   

       episode_score  episode_like  
54723           8.93             7  
54724           9.00             6  
54725           9.10            11  
54726           8.91             9  
54727           8.91            11

In [3]:
ls_all = df_episode[['titleid','episode_cnt']].values

In [4]:
episode_like = []
dates = []

In [18]:
# 회차별 좋아요
def Like_check(ls_all):
    count = 45988
    for i, ls in enumerate(ls_all[45988:]): 
        url = "https://comic.like.naver.com/likeIt/likeItContent.jsonp?_callback=window.__jindo2_callback._1399&serviceId=COMIC&displayId=COMIC&contentsId="+ str(ls[0]) + "_" + str(ls[1]) +"&lang=ko&viewType=like"
        print("Like check :: now : {}, titleid : {}, 회차 : {}".format(count + i,ls[0], ls[1]))

        response = requests.get(url)
        one = response.text.find("(") + 1  
        content = json.loads(response.text[one:-2])
        episode_like.append(content['result']['likeItContent']['likeItCount'])
        if i % 10 == 0:
            clear_output()

In [8]:
# 회차별 업데이트 시간(댓글 받을때 받자....)
def Date_check(ls):    
    url2 = "https://comic.naver.com/webtoon/detail.nhn?titleId=" + str(ls[0]) + "&no=" + str(ls[1])
    try:
        response = requests.get(url2)
        dom = BeautifulSoup(response.content, 'lxml')
        date = dom.select_one("#content > div.section_cont.wide > div.tit_area > div.vote_lst > dl.rt > dd.date").text
    #     print(date)
        dates.append(date)
    except:
        headers = {
            'cookie' : 'NNB=B5R2KGOJCPQVU; npic=z0dr3si6GPv0JLs+lOfKo07CMYF0ITC7YCWqlq0KbB4CdszDx1cgXjdJ9NYthOrdCA==; ASID=dc4cf77d0000016301ea74e50000005f; nx_ssl=2; JSESSIONID=FF8827E6E92420840A88E960BC2879B7; nid_inf=78978291; NID_AUT=JIDqQ1/dT5S8M6mX/CDHpJYiOh15MVnZNBZduCi+E2npkm9bixfXATPBVNNDy6tY; NID_JKL=tMUml42aRNANHk/yf3bq33lUR47W25egFSSOwQTR3Vk=; NID_SES=AAABojVIjCAaoRdiQlq8QR94oPsrDbqgBsRuq3HHaLXaF+MuPZIKW/mXdU8Indg2i2IONW9KulkpKmYk538igs764N7Z8OEr38urKjcUH+yeHhAHx100oTLIxvWE1GEJwg7fZtbVP+iJSEvhfXZFQ+75SDNziAxEhIsdUt4tMZKe5wpJiIG6MWfn+NQtJakeUFz52IKr+0y1WkTzezSBoFXrUYbMh9ZDjwCJrzyfW0E8qDS0I3DmEPTh97ngJLO7uq2Gv6Q/wavlraIynPoXZJ7ZW/qM2Bhoj6cSqq7s2E43aAqETTsHFEf69D7lGbTJRxrolClIdE5duha2x/Xom9qyB+h7uiifiyh2X9/1hxVS4S1zIs0YH8jw8gVgkXcSthzQ6LlaalboALHJeOoWe+vRLlObn+MkXj2zB31bJr+zADV3XOv0TkrWBN19M0XW1EWluHb1jnfGo4bM3FNm8VMpidOxfPj/hP/MJ08JVWppM5PoUPNjT6wCIfw36NKOasUU0FNu1/0waiLrsxs/dp+c/t8tSy+F+klJAs2HMuGHfgMPK+sohbghKHwyvwzZFUYPdA=='
        }
        response = requests.get(url2, headers=headers)
        dom = BeautifulSoup(response.content, 'lxml')
        date = dom.select_one("#content > div.section_cont.wide > div.tit_area > div.vote_lst > dl.rt > dd.date").text
        dates.append(date)

In [11]:
i = 10623
for ls in ls_all[10623:]:
    print("Date check :: now : {}, titleid : {}, 회차 : {}".format(i,ls[0], ls[1]))
    Date_check(ls)
    print("complete!")
    if i % 10 == 0:
        clear_output()
    i += 1

Date check :: now : 54721, titleid : 22073, 회차 : 7
complete!
Date check :: now : 54722, titleid : 22073, 회차 : 6
complete!
Date check :: now : 54723, titleid : 22073, 회차 : 5
complete!
Date check :: now : 54724, titleid : 22073, 회차 : 4
complete!
Date check :: now : 54725, titleid : 22073, 회차 : 3
complete!
Date check :: now : 54726, titleid : 22073, 회차 : 2
complete!
Date check :: now : 54727, titleid : 22073, 회차 : 1
complete!


In [12]:
len(dates)

54728

In [20]:
len(episode_like)

54728

In [22]:
df_episode['episode_like'] = episode_like
df_episode.to_csv("data_Juno/episode_per_webtoon_naver.csv", index=False)

In [14]:
df_episode['upload_date'] = dates
df_episode.tail()

title                                         image_link  \
54723    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54724    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54725    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54726    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   
54727    까뱅  https://shared-comic.pstatic.net/thumb/webtoon...   

                     episode  titleid  episode_cnt  total_comments  \
54723     GGAVANG 5화 <침투(2)>    22073            5             419   
54724     GGAVANG 4화 <침투(1)>    22073            4             384   
54725        GGAVANG 3화 <화산>    22073            3             507   
54726  GGAVANG 2화 <레이저 빔(2)>    22073            2             577   
54727  GGAVANG 1화 <레이저 빔(1)>    22073            1             945   

       episode_score  episode_like upload_date  
54723           8.93             7  2007.01.22  
54724           9.00             6  2007.01.22  
54725           9.10            11  2007.01.22  
54726           8.91             9  2007.01.22  
54727           8.91            11  2007.01.22

In [18]:
df_episode['upload_date'] = pd.to_datetime(df_episode['upload_date'])
df_episode.to_csv("data_Juno/episode_per_webtoon_naver.csv", index=False)

---
#### 웹툰 소개글 크롤링

In [2]:
import requests
from bs4 import BeautifulSoup

In [5]:
url = "https://comic.naver.com/webtoon/list.nhn?titleId=710747"
response = requests.get(url)
dom = BeautifulSoup(response.content, 'lxml')
dom.select_one("#content > div.comicinfo > div.detail > p").text

"재벌 3세가 일반 고등학교에 들어오면서, 선생들과 학생들은 거대 자본에 굴복해 권력을 차지하기 위한 '두뇌싸움'이 난무하는 학교로 바뀐다. 이 학교의 법칙을 부수기 위해 천재 '단건우'가 전학온다."

In [8]:
# 함수화
def get_intro(url):
    response = requests.get(url)
    dom = BeautifulSoup(response.content, 'lxml')
    return dom.select_one("#content > div.comicinfo > div.detail > p").text

# Restore the webtoon info 
info = pd.read_csv("data_Juno/webtoon_naver_info_by_genres_seperated.csv")

# Add Column
info['intro'] = info['main_link'].map(lambda x: get_intro(x))

info.intro

0                                  유미와 그녀의 우당탕탕 세포들 이야기.
1      황당하고 골 때리는 반전! \r\nTrauma(-정신적 외상)를 입을 만한 초특급 ...
2                  인기 웹툰 작가 총 출동\r\n마침내 시작되는 네이버 릴레이 웹툰!
3      무한도전X웹툰작가의 뒤 없이 달리는 릴레이 웹툰이\r\n방송과 함께 네이버 웹툰에서...
4      공감 100% 에 도전한다~ 절대 센스 Wony와 함께하는 골방 환상곡! 네이버 전...
5      재벌 3세가 일반 고등학교에 들어오면서, 선생들과 학생들은 거대 자본에 굴복해 권력...
6      우리 동네 편의점도 피해갈 수 없다!!\r\n편의점에서 일어나는 여러가지 에피소드들...
7      그녀가 했으니 당신도 할 수 있다!\r\n실패가 두렵지 않은 발칙한 요리 도전기, ...
8       로망이 꽃피는 캠퍼스는 없다.\r\n극사실주의에 기반한 너무나 현실적인 우리의 대학일기
9      와탕카='만세','지화자'\r\n좋은 패가 들어오면 외치는 단어 와탕카! 우주인 작...
10     생각하면 할 수록 무서운 빨간책 속 이야기. \r\n어디에나 있고 어디에도 없는 것...
11                            매주 새로운 주제의 6인 6색 만화를 즐기자! 
12                 금남의 공간 신비의 세계 여탕\r\n가본 자가 알려주는 그곳의 진실
13      심윤수 작가의 겨울 이야기.\r\n올 겨울을 따뜻하게 채워 줄 에피소드 12편.어...
14     섬세하고 자유분방한, 우유부단 하면서 분석력 있는\r\n서로 다른 혈액형들 사이에서...
15     우주인이 돌아왔다.\r\n업그레이드 된 작화와 센스 있는 유머. 와탕카 시즌2 시작합니다
16         동화를 객관적인 시각에서 보면 어떻게 될까?\r\n생소한 시각으로 동화 다시 보기
17     대를 물려 내려오는 고부갈등? 노노!\r\n

In [20]:
info2 = pd.read_csv("data_Juno/webtoon_naver_info_by_genres.csv")
info2.drop(columns=['non-free'], axis=1, inplace=True)
info2.tail()

unique_id      title   author  total_score  \
740     622643    진눈깨비 소년  쥬드프라이데이         9.97   
741     703840  한국만화 1990     웹툰작가         9.75   
742     336945    길에서 만나다  쥬드프라이데이         9.75   
743     515883  사또(Satto)      최윤진         9.80   
744     347685  슈퍼트리오 시즌2      황미나         7.36   

                                             main_link  \
740  https://comic.naver.com/webtoon/list.nhn?title...   
741  https://comic.naver.com/webtoon/list.nhn?title...   
742  https://comic.naver.com/webtoon/list.nhn?title...   
743  https://comic.naver.com/webtoon/list.nhn?title...   
744  https://comic.naver.com/webtoon/list.nhn?title...   

                                            image_link update_date  \
740  https://shared-comic.pstatic.net/thumb/webtoon...  2018.08.30   
741  https://shared-comic.pstatic.net/thumb/webtoon...  2018.05.17   
742  https://shared-comic.pstatic.net/thumb/webtoon...  2013.01.17   
743  https://shared-comic.pstatic.net/thumb/webtoon...  2015.12.29   
744  https://shared-comic.pstatic.net/thumb/webtoon...  2012.06.04   

                       genre  like_count  
740   story/pure/sensibility       76459  
741                    story       13787  
742  story/drama/sensibility        3564  
743      story/fantasy/drama       21366  
744        story/comic/drama          29

In [22]:
info2 = pd.concat([info2,info[['pay','episode_cnt','total_comments','episode_score','intro']]], axis=1)
info2.tail()

unique_id      title   author  total_score  \
740     622643    진눈깨비 소년  쥬드프라이데이         9.97   
741     703840  한국만화 1990     웹툰작가         9.75   
742     336945    길에서 만나다  쥬드프라이데이         9.75   
743     515883  사또(Satto)      최윤진         9.80   
744     347685  슈퍼트리오 시즌2      황미나         7.36   

                                             main_link  \
740  https://comic.naver.com/webtoon/list.nhn?title...   
741  https://comic.naver.com/webtoon/list.nhn?title...   
742  https://comic.naver.com/webtoon/list.nhn?title...   
743  https://comic.naver.com/webtoon/list.nhn?title...   
744  https://comic.naver.com/webtoon/list.nhn?title...   

                                            image_link update_date  \
740  https://shared-comic.pstatic.net/thumb/webtoon...  2018.08.30   
741  https://shared-comic.pstatic.net/thumb/webtoon...  2018.05.17   
742  https://shared-comic.pstatic.net/thumb/webtoon...  2013.01.17   
743  https://shared-comic.pstatic.net/thumb/webtoon...  2015.12.29   
744  https://shared-comic.pstatic.net/thumb/webtoon...  2012.06.04   

                       genre  like_count  pay  episode_cnt  total_comments  \
740   story/pure/sensibility       76459    0          224          201491   
741                    story       13787    0           20           10275   
742  story/drama/sensibility        3564    1           16           10191   
743      story/fantasy/drama       21366    1           12           20984   
744        story/comic/drama          29    1            6            1374   

     episode_score                                              intro  
740       9.963036  아무 일도 일어나지 않길 바랬던 고3 송해나는 우연히 정우진을 만난다그리고 우연은 ...  
741       9.697000                         1990년대 데뷔한, 인기 만화가들의 단편 특집  
742       9.746875                    서울 남산타워에서 우연히 만난 은희수와 미키의 로드무비.  
743       9.854167  인생 역전을 꿈꾸는 당신. 남은 인생을 걸어볼 용기가 있으신가요?\r\n복권 한 장...  
744       7.354000  대한민국 명작 만화 초대선.\r\n살아 있는 전설 황미나 작가의 명작 “슈퍼트리오”...

In [23]:
info2.to_csv("data_Juno/webtoon_naver_info_by_genres.csv", index=False)

In [35]:
info.rename({"title_x" : "title"}, axis=1, inplace=True)

In [36]:
info.to_csv("data_Juno/webtoon_naver_info_by_genres_seperated.csv", index=False)